library-compose/src/main/java/com/spendesk/grapes/compose/bucket/GrapesBucket.kt

```diff
@@ -0,0 +1,167 @@
+package com.spendesk.grapes.compose.bucket
+
+import androidx.compose.foundation.background
+import androidx.compose.foundation.border
+import androidx.compose.foundation.layout.Arrangement
+import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.PaddingValues
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.rememberScrollState
+import androidx.compose.foundation.verticalScroll
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.Immutable
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.clip
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.Dp
+import androidx.compose.ui.unit.dp
+import com.spendesk.grapes.compose.theme.GrapesTheme
+
+/**
+ * @author jean-philippe
+ * @since 15/03/2023, Wednesday
+ **/
+
+@Immutable
+object GrapesBucketDefaults {
+    private val BorderWidth = 0.2.dp
+    private val BucketPadding = 16.dp
+
+    @Composable
+    fun DefaultGrapesBucketDecorationBox(
+        modifier: Modifier = Modifier,
+        contentPaddingValues: PaddingValues = bucketPadding(),
+        innerContent: @Composable () -> Unit
+    ) {
+        Box(
+            modifier = modifier
+                .background(GrapesTheme.colors.mainWhite, GrapesTheme.shapes.medium)
+                .border(BorderWidth, GrapesTheme.colors.mainNeutralNormal, GrapesTheme.shapes.medium)
+                .clip(GrapesTheme.shapes.medium)
+                .padding(contentPaddingValues)
+        ) {
+            innerContent()
+        }
+    }
+
+    fun bucketPadding(
+        start: Dp = BucketPadding,
+        top: Dp = BucketPadding,
+        end: Dp = BucketPadding,
+        bottom: Dp = BucketPadding
+    ): PaddingValues = PaddingValues(start, top, end, bottom)
+}
+
+@Composable
+fun GrapesBucket(
+    title: String,
+    modifier: Modifier = Modifier,
+    action: String? = null,
+    actionColor: Color? = null,
+    onActionClicked: (() -> Unit)? = null,
+    decorationBox: @Composable (innerContent: @Composable () -> Unit) -> Unit =
+        @Composable { innerContent ->
+            GrapesBucketDefaults.DefaultGrapesBucketDecorationBox(innerContent = innerContent)
+        },
+    content: @Composable (() -> Unit)? = null
+) {
+    val headLine = @Composable {
+        GrapesBucketHeadline(
+            title = title,
+            action = action,
+            actionColor = actionColor,
+            onActionClicked = onActionClicked
+        )
+    }
```

Comment:
Open question: why do we need to use this syntax here: `@Composable { ... }`?
The `GrapesBucketHeadline` is already a @Composable, so i'm wondering if could just do that instead:
```suggestion
    val headLine =
        GrapesBucketHeadline(
            title = title,
            action = action,
            actionColor = actionColor,
            onActionClicked = onActionClicked
        )
```
---
I've seen that a few times in this PR, so I only write one comment for all :)

Reply:
it doesn't work without this syntax, otherwise, I had to wrap the headline val in a lambda `{}` like
```
GrapesBucket(
        modifier = modifier,
        decorationBox = decorationBox,
        headline = { headLine } ,
        content = content
    )
```

this `@Composable {}` annotation prevent this

Reply:
GrapesBucketHeadline is a Composable, but Composable invocation can only happen from the context of a Conposable function / expression lambda (otherwise your IDE will show you an error), so I must add the @Composable annotation.

Reply:
Thanks!